profitbricks/src/test/java/org/jclouds/profitbricks/features/IpBlockApiMockTest.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.features;
+
+import com.squareup.okhttp.mockwebserver.MockResponse;
+import com.squareup.okhttp.mockwebserver.MockWebServer;
+import java.util.List;
+import org.jclouds.profitbricks.ProfitBricksApi;
+import org.jclouds.profitbricks.domain.IpBlock;
+import org.jclouds.profitbricks.domain.Location;
+import org.jclouds.profitbricks.internal.BaseProfitBricksMockTest;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import org.testng.annotations.Test;
+
+@Test(groups = "unit", testName = "IpBlockApiMockTest")
+public class IpBlockApiMockTest extends BaseProfitBricksMockTest {
+
+   @Test
+   public void testGetOneIpBlock() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/ipblock/ipblock.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      IpBlockApi api = pbApi.ipBlockApi();
+
+      String id = "qwertyui-qwer-qwer-qwer-qwertyyuiiop";
+
+      String content = "<ws:getPublicIpBlock><blockId>" + id + "</blockId></ws:getPublicIpBlock>";
+
+      try {
+         IpBlock ipBlock = api.getIpBlock(id);
+         assertRequestHasCommonProperties(server.takeRequest(), content);
+         assertNotNull(ipBlock);
+         assertEquals(ipBlock.id(), id);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetNonExisingIpBlock() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setResponseCode(404));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      IpBlockApi api = pbApi.ipBlockApi();
+
+      String id = "qwertyui-qwer-qwer-qwer-qwertyyuiiop";
+
+      try {
+         IpBlock ipBlock = api.getIpBlock(id);
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNull(ipBlock);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+
+   @Test
+   public void testGetAllIpBlock() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue(new MockResponse().setBody(payloadFromResource("/ipblock/ipblocks.xml")));
+
+      ProfitBricksApi pbApi = api(server.getUrl(rootUrl));
+      IpBlockApi api = pbApi.ipBlockApi();
+
+      try {
+         List<IpBlock> ipBlocks = api.getAllIpBlock();
+         assertRequestHasCommonProperties(server.takeRequest());
+         assertNotNull(ipBlocks);
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
```

Comment:
The get all blocks method also defines a 404 fallback. Add the corresponding test.